### DISCUSSION TITLE
Should a pure Server Component page have no JavaScript?

### DISCUSSION THREAD

**Discussion by stigkj** [vercel/next.js#30550] [App Router] [answered]
After reading the introduction to Server Components, specifically this paragraph:

> With Server Components, there's zero client-side JavaScript needed, making page rendering faster

I got the impression that a pure Server Component page should load no JavaScript. But after trying the demo and adding a page that was completely static, it still loaded JavaScript in the production build.

Is this something that is still to be done, or is there something I have not understood correctly?

**Comment by stigkj** [comment, marked as answer]
Shouldn't this be possible @leerob?

**Reply by leerob** [reply]
Right now, it's still experimental and we're working with the React team to work on these APIs. Everything is subject to change - but you are correct there is a small amount of JavaScript used to replace the suspended data you're fetching.

**Reply by stigkj** [reply]
Sounds good 👍

**Reply by madjam002** [reply]
Javascript is even required to render the page at all, so if JS is disabled the page is completely blank, even without the use of React Suspense. This appears to be because all of the server HTML is wrapped in various `<template />` elements which I assume is needed for the current implementation of Server Components.

It would be good if even when JS is disabled the page could be rendered in its initial state (obviously any suspended trees would then remain suspended indefinitely because the JS can't run to stream the loaded states).

Obviously it's all experimental at the moment but that's just some feedback from having played around with it.
